src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -301,6 +301,9 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
 
+	/* we will not continue if there is a version mismatch */
+	ErrorIfInstalledVersionMismatch();
```

Comment:
This is fairly late though, at that point we'll already have written to the catalog and such (which could cause issues w/ a mismatch).  That's not sufficient - I think this needs to be around the places where we currently check EnsureCoordinator();